sw/dnn/layernorm/verify.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+# Copyright 2023 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Luca Colagrande <colluca@iis.ee.ethz.ch>
+
+import sys
+from pathlib import Path
+import numpy as np
+import torch
+from data.datagen import golden_model
+
+sys.path.append(str(Path(__file__).parent / '../../../util/sim/'))
+import verification  # noqa: E402
+from elf import Elf  # noqa: E402
+from data_utils import bytes_to_float, bytes_to_struct  # noqa: E402
+
+
+ERR_THRESHOLD = 0.001
+
+PRECISION_T = {
+    8: '64',
+    4: '32',
+    2: '16',
+    1: '8'
+}
+
+NUMPY_T = {
+    '64': np.float64,
+    '32': np.float32,
+    '16': np.float16
+}
+
+
+def main():
+    # Run simulation and get outputs
+    args = verification.parse_args()
+    raw_results = verification.simulate(sim_bin=args.sim_bin,
+                                        snitch_bin=args.snitch_bin,
+                                        symbols_bin=args.symbols_bin,
+                                        log=args.log,
+                                        output_uids=['ofmap'])
+
+    # Extract input operands from ELF file
+    if args.symbols_bin:
+        elf = Elf(args.symbols_bin)
+    else:
+        elf = Elf(args.snitch_bin)
+
+    layer_struct = {
+        'batch_size': 'I',
+        'seq_len': 'I',
+        'embeddings': 'I',
+        'n_tiles': 'I',
+        'eps': 'f',
+        'ifmap_ptr': 'I',
+        'ofmap_ptr': 'I',
+        'dtype': 'I'
+    }
+    layer = bytes_to_struct(elf.get_symbol_contents('layer'), layer_struct)
+    batch_size = layer['batch_size']
+    seq_len = layer['seq_len']
+    embeddings = layer['embeddings']
+    eps = layer['eps']
+    prec = PRECISION_T[layer['dtype']]
+
+    ifmap = np.array(bytes_to_float(elf.get_symbol_contents('ifmap'), prec), dtype=NUMPY_T[prec])
```

Comment:
In general, we could replace this with [`np.frombuffer`](https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html) in all verify scripts. See [here](https://github.com/pulp-platform/snitch_cluster/blob/9b6cd8de64212661c5fc3bb39dc952ce159980d9/sw/blas/gemm/verify.py#L57) for an example

Reply:
Leaving this to @colluca

Reply:
Saw this, definitely a good idea to use the numpy functions. I would however wrap them in our own custom function, such that we can handle `fp8` precision with the same function as well.